access_control.py

```diff
@@ -0,0 +1,225 @@
+    print(f"RegisterCI command: {register_ci_cmd_str}")
+    subprocess.check_call(register_ci_cmd_str,
+        shell=True,
+    )
+
+
+def check_if_repo_exists(feedstock_name):
+    repo = f"{feedstock_name}-feedstock"
+    owner_repo = f"{GH_ORG}/{repo}"
+    print(f"Checking if {owner_repo} exists")
+    response = requests.get(f"https://api.github.com/repos/{owner_repo}")
+    if response.status_code != 200:
+        raise ValueError(f"Repository: {owner_repo} not found!")
+
+def _check_for_path(path):
+    mapping = _get_filename_feedstock_mapping(path)
+    for filename, feedstocks in mapping.items():
+        print(f"Checking if {filename} is in {CIRUN_FILENAME_RESOURCE_MAPPING.keys()}")
+        assert filename in CIRUN_FILENAME_RESOURCE_MAPPING
+        for feedstock in feedstocks:
+            check_if_repo_exists(feedstock)
+    if not mapping:
+        print(f"Nothing to check for: {path}")
+
+def check():
+    """Check requests are valid (resources exist, feedstocks exist)"""
+    _check_for_path("grant_access")
+    _check_for_path("revoke_access")
+
+def _commit_after_files_removal(push=True):
+    subprocess.check_call(
+        "git add .",
+        shell=True,
+    )
+
+    subprocess.check_call(
+            "git commit --allow-empty -am 'Remove access control files'",
+            shell=True,
+        )
+    if push:
+        subprocess.check_call(
+            "git push",
+            shell=True,
+        )
+
+
+def _remove_input_files(dir, file_to_keep):
+    print("Removing input files")
+    directory = Path(dir)
+    files_to_keep = [Path(file_to_keep)]
+    if not directory.is_dir():
+        raise ValueError("The specified path is not a directory.")
+
+    for file in directory.iterdir():
+        if file.is_file() and file not in files_to_keep:
+            print(f"Removing input file: {file}")
+            file.unlink()
+    _commit_after_files_removal(push=False)
+
+
+def update_access_yaml(resource, feedstock_name, action, filename=ACCESS_YAML_FILENAME):
+    """
+    Manage feedstock in .access_control.yml.
+
+    Parameters:
+    - resource: the access control resource (e.g., "travis", "cirun-gpu-small").
+    - feedstock_name: the name of the feedstock.
+    - action: either "add" or "remove".
+    - filename: the name of the YAML file.
+    """
+    print(f"Updating {filename}")
+    yaml = YAML()
+    yaml.indent(mapping=2, sequence=4, offset=2)  # Indent settings for 2-space mapping
+
+    with open(filename, 'r') as f:
+        content = yaml.load(f)
+
+    # If resource doesn't exist, create it
+    if resource not in content['access_control']:
+        content['access_control'][resource] = []
+
+    if action == "add":
+        # Check if feedstock already exists
+        if any(entry["feedstock"] == feedstock_name for entry in content['access_control'][resource]):
+            print(f"Feedstock {feedstock_name} already exists under {resource}. Skipping addition.")
```

Comment:
Is there a scenario where a feedstock can be added to the same resource twice, maybe with different policies? 🤔 I cannot think of anything but double checking.